### PR TITLE
std.math.hardware: Fix LoongArch support

### PR DESCRIPTION
**Pull request by heiher**
This PR fixes compile error and adds cases for LDC.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
I think what you actually want to do here is submit a PR to `ldc` proper that update the `runtime/phobos` git sumbodule to point it to this commit. But I can never remember that works.

**Comment by heiher** [comment]
> I think what you actually want to do here is submit a PR to `ldc` proper that update the `runtime/phobos` git sumbodule to point it to this commit. But I can never remember that works.

I submitted the [`movfcsr2gr` part](https://github.com/dlang/phobos/pull/9000) to the upstream repo `dlang/phobos` and the rest here. The overlapping parts will be merged in the next update of the `phobos` submodule. Thanks.

File: std/math/hardware.d [hunk 1]
```diff
@@ -1049,6 +1059,11 @@ private:
             {
                 asm pure nothrow @nogc { "frcsr %0" : "=r" (cont); }
             }
+            else version (LoongArch_Any)
+            {
+                asm pure nothrow @nogc { "movfcsr2gr %0, $fcsr0" : "=r" (cont); }
+                cont &= (roundingMask | allExceptions);
```
Comment (kinke) [inline std/math/hardware.d:1065]: Hi, thanks for the PRs! Looks reasonable to me, except for this line here - is the masking really required here for this internal function? We don't mask for the other ISAs.

Comment (heiher) [reply std/math/hardware.d:1065]: It seems so. This is copied from https://github.com/ldc-developers/phobos/pull/73/files#diff-a76c79337d2c4e1daf6668e751145248ef1b49cea7f56ac1fbb39e4d6e0b05acL1093

Thanks.

Comment (kinke) [reply std/math/hardware.d:1065]: Oh I see, some ugly duplication - about time to finally upstream this stuff.

**Review by kinke — APPROVED** [review]

**Pull request merged**